lib/tictactoe.rb

```diff
@@ -8,35 +8,55 @@ def initialize(io:)
   end
 
   def render_board
-    io.puts "#{@board[0]}|#{@board[1]}|#{@board[2]}"
-    io.puts '------'
-    io.puts "#{@board[3]}|#{@board[4]}|#{@board[5]}"
-    io.puts '------'
-    io.puts "#{@board[6]}|#{@board[7]}|#{@board[8]}"
+    board_output = <<~BOARD_RENDER 
+    #{@board[0]}|#{@board[1]}|#{@board[2]}
+    ------
+    #{@board[3]}|#{@board[4]}|#{@board[5]}
+    ------
+    #{@board[6]}|#{@board[7]}|#{@board[8]}
+    BOARD_RENDER
+    io.puts board_output
     board
   end
 
   def place_token(position)
     board[position] = current_token
+    switch_turn
     render_board
   end
 
   def current_player
-    @current_token
+    current_token
   end
 
   def begin_player_turn
+    select_another_position_prompt = 'Select another position to place your token [1 - 9]: '
     desired_position = io.gets.to_i
     io.puts 'Select a position to place your token [1 - 9]: '
 
-    if board[desired_position] == ' '
+    list_of_accepted_inputs = [*1..9]
+    if !list_of_accepted_inputs.include? desired_position
+      io.puts 'Invalid position'
+      io.puts select_another_position_prompt
+      render_board
+    elsif board[desired_position] == ' '
       place_token(desired_position)
     else
       io.puts 'That position is taken!'
-     end
+      io.puts select_another_position_prompt
+      render_board
+    end
   end
 
   private
 
   attr_accessor :board, :current_token, :io
+
+  def switch_turn
+    if @current_token == 'X'
```

Comment:
Yeah. So, apparently, you can reference an `attr` variable without the `@` or `self.` if you're just referencing it because if Ruby can't find it in the local scope, it will look for methods with that name in the instance or class scope.

For _assigning_ a variable, Ruby always assumes it's local. If it's a method (like `name=`), Ruby ignores it. I tried just doing the equality check in `#switch_turn` with `current_token == 'X'` instead of `@current_token == 'X'` and it returns the correct value. But when I try to set `current_token` without the `@` Ruby doesn't set the value. 

This was supper surprising so I checked the other parts of Tic Tac Toe and realized the only time I reference an attr_accessor variable it's just getting the value, it's not setting it. That's why I never got an error for `#current_player`. It only returns `current_token`.